compiler/AST/AggregateType.cpp

```diff
@@ -2224,9 +2226,124 @@ void AggregateType::buildDefaultInitializer() {
   }
 }
 
+static bool hasFullyGenericField(AggregateType* at) {
+  for_fields(field, at) {
+    DefExpr* defExpr = field->defPoint;
+    if (!field->hasFlag(FLAG_TYPE_VARIABLE) && !field->hasFlag(FLAG_PARAM) &&
+        !field->hasFlag(FLAG_SUPER_CLASS)) {
+      if (defExpr->exprType == nullptr && defExpr->init == nullptr) {
+        return true;
+      }
+    }
+  }
+
+  if (at->isClass()                    ==  true &&
+      at->symbol->hasFlag(FLAG_REF)    == false &&
+      at->dispatchParents.n            >      0 &&
+      at->symbol->hasFlag(FLAG_EXTERN) == false) {
+    if (AggregateType* parent = at->dispatchParents.v[0]) {
+      return hasFullyGenericField(parent);
+    }
+  }
+
+  return false;
+}
+
+void AggregateType::buildReaderInitializer() {
+  if (!fUseIOFormatters) return;
+  if (this->getModule()->modTag == MOD_INTERNAL) return;
```

Comment:
Looks like a workaround / limitation for now. If it's a workaround, let's put a comment identifying it as such.

Reply:
A limitation due to the unavailability of ``fileReader`` in the internal modules. I'll add a comment.